worker/caasmodeloperator/modeloperator.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caasmodeloperator
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+	"github.com/juju/utils"
+	"github.com/juju/version"
+	"gopkg.in/juju/worker.v1/catacomb"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api/caasmodeloperator"
+	"github.com/juju/juju/caas"
+)
+
+// ModelOperatorBroker describes the caas broker interface needed for installing
+// a ModelOperator into Kubernetes
+type ModelOperatorBroker interface {
+	EnsureModelOperator(string, string, *caas.ModelOperatorConfig) error
+}
+
+// ModelOperatorManager defines the worker used for managing model operators in
+// caas
+type ModelOperatorManager struct {
+	agentConfig agent.Config
+	api         caasmodeloperator.Interface
+	broker      ModelOperatorBroker
+	catacomb    catacomb.Catacomb
+	logger      Logger
+	modelUUID   string
+}
+
+const (
+	// DefaultModelOperatorPort is the default port used for the api server on
+	// the model operator
+	DefaultModelOperatorPort = 3300
+)
+
+// Kill implements worker kill method
+func (m *ModelOperatorManager) Kill() {
+	m.catacomb.Kill(nil)
+}
+
+func (m *ModelOperatorManager) loop() error {
+	m.logger.Debugf("gathering model operator provisioning information for model %s", m.modelUUID)
+	info, err := m.api.ModelOperatorProvisioningInfo(m.modelUUID)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	password, err := utils.RandomPassword()
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	errResults, err := m.api.SetPasswords([]caasmodeloperator.CAASModelPassword{
+		caasmodeloperator.CAASModelPassword{
+			Model:    m.modelUUID,
+			Password: password,
+		},
+	})
```

Comment:
Only need to set the password if the model operator does not exist.
See the logic in the application operator worker

Reply:
Looking at this now